cassiopeia/solver/DistanceSolver.py

```diff
@@ -75,6 +75,40 @@ def __init__(
         self.dissimilarity_function = dissimilarity_function
         self.add_root = add_root
 
+    def get_dissimilarity_map(
+        self, 
+        cassiopeia_tree: CassiopeiaTree,
+        layer: Optional[str] = None
+    ) -> pd.DataFrame: 
+        """Obtains or generates a matrix that is updated throughout the solver.
+
+        The highest-level method to obtain a dissimilarity map, which
+        will be the matrix primarily used throughout the solve method. This
+        matrix contains the pairwise dissimilarity between samples which is used
+        for identifying sample pairs to merge, and will be updated at every
+        iteration within the solve method. This method is not limited to
+        outputting dissimilarity maps, but is instead deliberately designed to
+        be overwritten to allow for similarity maps and lambda matrices if
+        needed.
```

Comment:
The idea of a "lambda" matrix is specific to the spectral neighbor joining algorithm. I'd prefer if the description ended with something like:

"... but is instead deliberately designed to be overwritten to allow for similarity maps or other algorithm-specific sample-sample comparison maps in derived classes."